### PR TITLE
Refactor Cassandra queue persistence for later NoSQL support

### PR DESCRIPTION
**Pull request by longquanzheng**
**What changed?**
Refactor Cassandra queue persistence 

**Why?**
For later NoSQL support in https://github.com/uber/cadence/issues/3514

**How did you test it?**
Existing persistence tests

**Potential risks**
Should be minimal risk as only refactoring code, no behavior change

### CODE REVIEW

File: common/persistence/nosql/nosqlplugin/cassandra/queue.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+		payload := getMessagePayload(message)
+		id := getMessageID(message)
+		rows = append(rows, nosqlplugin.QueueMessageRow{ID: id, Payload: payload})
+		message = make(map[string]interface{})
+	}
+
+	nextPageToken := iter.PageState()
+	// TODO? can we remove the below code 
+	//newPageToken := make([]byte, len(nextPageToken))
+	//copy(newPageToken, nextPageToken)
```
Comment (longquanzheng) [inline common/persistence/nosql/nosqlplugin/cassandra/queue.go:137]: @meiliang86 the existing code copy the token to a new slice, I don't think that's necessary, but let me know if I missed something.

File: common/persistence/cassandra/cassandraQueue.go [hunk 2]
```diff
@@ -102,392 +71,300 @@ func newQueue(
...
 	// Use negative queue type as the dlq type
-	query := q.session.Query(templateDeleteMessageQuery, q.getDLQTypeFromQueueType(), messageID)
-	if err := query.Exec(); err != nil {
-		return &workflow.InternalServiceError{
-			Message: fmt.Sprintf("DeleteMessageFromDLQ operation failed. Error %v", err),
-		}
-	}
-
-	return nil
+	return q.db.DeleteMessage(ctx, q.queueType, messageID)
```
Comment (yux0) [inline common/persistence/cassandra/cassandraQueue.go:239]: use q.getDLQTypeFromQueueType()

File: common/persistence/cassandra/cassandraQueue.go [hunk 3]
```diff
@@ -102,392 +71,300 @@ func newQueue(
...
 	// Use negative queue type as the dlq type
-	query := q.session.Query(templateDeleteMessagesQuery, q.getDLQTypeFromQueueType(), firstMessageID, lastMessageID)
-	if err := query.Exec(); err != nil {
-		return &workflow.InternalServiceError{
-			Message: fmt.Sprintf("RangeDeleteMessagesFromDLQ operation failed. Error %v", err),
-		}
-	}
-
-	return nil
+	return q.db.RangeDeleteMessages(ctx, q.queueType, firstMessageID, lastMessageID)
```
Comment (yux0) [inline common/persistence/cassandra/cassandraQueue.go:248]: q.getDLQTypeFromQueueType()

File: common/persistence/cassandra/cassandraQueue.go [hunk 4]
```diff
@@ -102,392 +71,300 @@ func newQueue(
...
-		lastMessageID,
-	).PageSize(pageSize).PageState(pageToken)
-
-	iter := query.Iter()
-	if iter == nil {
-		return nil, nil, &workflow.InternalServiceError{
-			Message: fmt.Sprintf("ReadMessagesFromDLQ operation failed. Not able to create query iterator."),
-		}
+	response, err := q.db.GetMessagesBetween(ctx, nosqlplugin.GetMessagesBetweenRequest{
+		QueueType:               q.queueType,
```
Comment (yux0) [inline common/persistence/cassandra/cassandraQueue.go:205]: q.getDLQTypeFromQueueType()

**Comment by longquanzheng** [comment]
@yux0 I have addressed the comments (fixed the bug). Thanks for review.

**Review by yux0 — APPROVED** [review]

**Pull request merged**